kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/TlsSecrets.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+
+public record TlsSecrets(Path path, String volumeName, String secretName) {
+
+    private static final Path TLS_SECRETS_ROOT_DIR = Path.of("/opt/kroxylicious/secrets/tls");
+    private static final String CERTIFICATE = "tls.crt";
+    private static final String KEY = "tls.key";
+
+    public static List<TlsSecrets> tlsSecretsFor(KafkaProxy proxy) {
+        return proxy.getSpec().getListeners().stream().flatMap(listeners -> listeners.getTls().getCertificateRefs().stream().map(cert -> {
+            Path resolve = TLS_SECRETS_ROOT_DIR.resolve(listeners.getName()).resolve(cert.getName());
+            return new TlsSecrets(resolve, "tls-secrets-" + listeners.getName() + "-" + cert.getName(), cert.getName());
```

Comment:
should probably not use user input here since we could build a name longer than the limit, by the spec it looks like volumeName should be a DNS label (max 63 chars). We are iterating in a consistent order so we could name them like `tls-secrets-0`, `tls-secrets-1`.